[Autocomplete] Fix unexpected clearing

SerhiiBilyk opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/master/CONTRIBUTING.md#sending-a-pull-request).

Closes #18646

SerhiiBilyk commented on docs/src/pages/components/autocomplete/Tags.js:
```diff
@@ -58,6 +58,8 @@ export default function Tags() {
         options={top100Films.map(option => option.title)}
         defaultValue={[top100Films[13].title]}
         freeSolo
+        autoHighlight
```
It is only for dev and test purpose. I'll remove it once PR will be ready for unit tests.

oliviertassinari replied:
Why changing the demos?

SerhiiBilyk commented:
@eps1lon sure.

SerhiiBilyk commented:
@oliviertassinari @eps1lon 
Is there a precedense of useAutocomplete props?
I mean if for example all `[autoSelect, freeSolo, multiple, autoHighlight]` are true.
Then I type only one char `a`, then press `Arrow down`, then `blur`.
Input value should be `a` or the first highligted row or empty value ?

It is hard to figure out the logic (predict all usecases).
I've spent several hours and my code looks like: 
![bug](https://user-images.githubusercontent.com/16646733/73611467-69e8a080-45e2-11ea-885c-a464406163a9.gif)

oliviertassinari commented:
> Is there a precedense of useAutocomplete props?
I mean if for example all [autoSelect, freeSolo, multiple, autoHighlight] are true.
Then I type only one char a, then press Arrow down, then blur.
Input value should be a or the first highligted row or empty value ?

@SerhiiBilyk You can assume the current behavior on master is correct.

> Could you please also explain me what does mean next error:

This sounds like an installation issue on your environment.

SerhiiBilyk commented:
@oliviertassinari @eps1lon Could you please check ?

SerhiiBilyk commented:
> * I have updated the test case to be failing without the patch. Otherwise, it won't prevent future regressions. This is especially important as a case I have missed during the initial implementation.
> * They are two different problems solved at once here. I have added a new test case.

Good point!
Thanks for the help

oliviertassinari commented:
@SerhiiBilyk Thank youç
